--- tools/v8_gypfiles/v8.gyp
+++ tools/v8_gypfiles/v8.gyp
@@ -1368,9 +1368,14 @@
     },  # cppgc_base
     {
       'target_name': 'v8_cppgc_shared',
-      'type': 'none',
+      'type': 'static_library',
       'toolsets': ['host', 'target'],
-      'direct_dependent_settings': {
+      # 'direct_dependent_settings': {
+        'conditions': [
+          ['enable_lto=="true"', {
+            'cflags_cc': [ '-fno-lto' ],
+          }],
+        ],
         'sources': [
           '<(V8_ROOT)/src/heap/base/stack.cc',
           '<(V8_ROOT)/src/heap/base/stack.h',
@@ -1447,7 +1452,7 @@
             ],
           }],
         ],
-      },
+      # },
     },  # v8_cppgc_shared
 
     ###############################################################################
